src/app/centers/create-center/create-center.component.ts

```diff
@@ -0,0 +1,147 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, FormControl, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { GroupsService } from 'app/groups/groups.service';
+import { CentersService } from '../centers.service';
+
+/**
+ * Create Center component.
+ */
+@Component({
+  selector: 'mifosx-create-center',
+  templateUrl: './create-center.component.html',
+  styleUrls: ['./create-center.component.scss']
+})
+export class CreateCenterComponent implements OnInit {
+
+  /** Minimum date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum date allowed. */
+  maxDate = new Date();
+  /** Center form. */
+  centerForm: FormGroup;
+  /** Office data. */
+  officeData: any;
+  /** Group data. */
+  groupsData: any;
+  /** Staff data. */
+  staffData: any;
+  /** Group Members. */
+  groupMembers: any[] = [];
+  /** Group Choice. */
+  groupChoice = new FormControl('');
+
+  /**
+   * Retrieves the offices data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {CentersService} centerService CentersService.
+   * @param {GroupsService} groupService GroupsService.
+   * @param {DatePipe} datePipe Date Pipe to format date.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private router: Router,
+              private centerService: CentersService,
+              private groupService: GroupsService,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe( (data: { offices: any }) => {
+      this.officeData = data.offices;
+    });
+  }
+
+  /**
+   * Creates and sets the center form.
+   */
+  ngOnInit() {
+    this.createCenterForm();
+  }
+
+  /**
+   * Creates the center form.
+   */
+  createCenterForm() {
+    this.centerForm = this.formBuilder.group({
+      'name': ['', [Validators.required, Validators.pattern('(^[A-z]).*')]],
+      'officeId': ['', Validators.required],
+      'submittedOnDate': [''],
+      'staffId': [''],
+      'externalId': [''],
+      'active': [''],
+    });
+    this.buildDependencies();
+  }
+
+  /**
+   * Sets the staff and groups data each time the user selects a new office.
+   * Adds form control Activation Date if active.
+   */
+  buildDependencies() {
+    this.centerForm.get('officeId').valueChanges.subscribe( (option: any) => {
+      this.groupService.getGroupsByOfficeId(option).subscribe( (data: any) => {
+        this.groupsData = data;
+        if (!this.groupsData.length) {
+          this.groupChoice.disable();
+        }
```

Comment:
Add an else condition here to enable the same when the groupData has a length greater than 0. These are the endpoints for the test, as if a user initially selects a officeId having groupData of length 0, and then changes it to another officeId having groupData greater than 0, the field will remain disabled.